primo/opt_model/opt_model.py

```diff
@@ -419,6 +427,38 @@ def project_max_spend_rule(model, campaign_id):
                 doc="Define limit on max project spend",
             )
             LOGGER.info("Added max project spend constraint")
+
+        def budget_constraint_slack(model):
+            """
+            Implement a slack variable to maximize the utilization of the total budget.
```

Comment:
"Implement a..." -> "Implements a constraint that calculates the amount of budget available that does not get utilized"